packages/charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -0,0 +1,186 @@
+import * as React from 'react';
+import { max as d3Max } from 'd3-array';
+import { axisLeft as d3AxisLeft, axisBottom as d3AxisBottom, Axis as D3Axis } from 'd3-axis';
+import { scaleBand as d3ScaleBand, scaleLinear as d3ScaleLinear, ScaleLinear as D3ScaleLinear } from 'd3-scale';
+import { select as d3Select } from 'd3-selection';
+import { classNamesFunction, customizable, IClassNames } from '../../Utilities';
+import {
+  IHorizontalBarChartProps,
+  IHorizontalBarChartStyleProps,
+  IHorizontalBarChartStyles,
+  IDataPoint
+} from './HorizontalBarChart.types';
+
+const getClassNames = classNamesFunction<IHorizontalBarChartStyleProps, IHorizontalBarChartStyles>();
+type numericAxis = D3Axis<number | { valueOf(): number }>;
+type stringAxis = D3Axis<string>;
+
+@customizable('HorizontalBarChart', ['theme', 'styles'])
+export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartProps, {}> {
+  private _points: IDataPoint[];
+  private _width: number;
+  private _height: number;
+  private _barHeight: number;
+  private _yAxisTickCount: number;
+  private _colors: string[];
+  private _classNames: IClassNames<IHorizontalBarChartStyles>;
```

Comment:
Any data that affects rendering should not be a class member. It should be part of props or state. And I think these are all props with default values. You can define a static class member `defaultProps` that React will automatically use to populate default values. You can search `defaultProps` for examples like this:

```
  public static defaultProps: Partial<ICoachmarkTypes> = {
    isCollapsed: true,
    mouseProximityOffset: 10,
    delayBeforeMouseOpen: 3600, // The approximate time the coachmark shows up
    color: DefaultPalette.themePrimary,
    isPositionForced: true,
    positioningContainerProps: {
      directionalHint: DirectionalHint.bottomAutoEdge
    }
  };
```

Any member affecting rendering that is modified internally (by something other than just a default value) should be part of component state, not a class member.

Reply:
(This comment excludes `_classNames` which usually is a class member updated when prop/state changes.)